### PR TITLE
Interview - Route service functions

### PR DESCRIPTION
**Pull request by pankajsy**
Java class file for implementing route services

### CODE REVIEW

File: HHinterview.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Scanner;
+import java.util.Set;
+
+
+/*Intsructions for running:
```
Comment (brentnd) [inline HHinterview.java]: These should be in the README file under Build instructions.

File: HHinterview.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+Unique routes: ["SRT", "CVT", "Perkiomen", "Welsh Mountain", "Oaks to Philly"]
+For user 42: ["42SRT", "42CVT", "42Perkiomen", "CVT42", "Perkiomen42", "Welsh Mountain42", "42SRT42", "42Welsh Mountain42", "42Oaks to Philly42"]
+For user 42 services ["Komoot", "RWGPS"]: ["42SRT42", "42Welsh Mountain42", "42Oaks to Philly42", "CVT42", "Perkiomen42", "Welsh Mountain42"]
+*/
+
+/**
+ * The HHinterview is a class that has implementations
+ * for three "services" (Strava, RWGPS, and Komoot).
+ */
+public class HHinterview {
```
Comment (brentnd) [inline HHinterview.java]: How would you refactor this class if we wanted to add or remove services dynamically without modifying the base `HHinterview` class?

File: HHinterview.java (hunk 1 above)
Comment (pankajsy) [reply HHinterview.java]: Got it. I'll make that change.

File: HHinterview.java (hunk 2 above)
Comment (pankajsy) [reply HHinterview.java]: In order to achieve each service have it's own implementation, I'm thinking to maintain a Service Class, fields 'name' and 'routes' with getter/setter for the same. 

To add/remove services dynamically, we can maintain a List of Service Objects in HHinterview Class. 

Rest of the functions shall change a bit, now that we have List of Service Objects.

**Comment by brentnd** [comment]
@pankajsy thanks for the submission and the follow up to the review.

Closing this and we will follow up with you.

**Comment by pankajsy** [comment]
@brentnd I was hoping to push the refactored object oriented code as mentioned yesterday, by end of today. I just commented yesterday but did not push the final changes yet.

**Comment by brentnd** [comment]
@pankajsy my mistake, I didn't realize you were still working on more changes. Feel free to push when you have them.

**Pull request closed without merging**